### PR TITLE
add istio and cloudrun GKE features

### LINKED ISSUE #2649: Support for google_container_cluster istio addon
Now that istio is released in beta https://cloud.google.com/blog/products/containers-kubernetes/accelerate-your-app-delivery-with-kubernetes-and-istio-on-gke

And that it can be added as an addon:

screen shot 2018-12-13 at 9 19 14 am

We should add it as an add-on to the google beta provider: for example:
```
resource "google_container_cluster" "foo" {
  name = "foo"
[  ....]
  addons_config {
    istio {
      disabled = false
    }
  }
}
```

### PR DESCRIPTION
**Pull request by pondohva**
fixes #2649

### CODE REVIEW

**Comment by pondohva** [comment]
Do I need to change base branch to 2.0.0?

**Comment by danawillow** [comment]
Yes please!

File: google/resource_container_cluster.go [hunk 1]
```diff
@@ -1490,6 +1525,23 @@ func expandClusterAddonsConfig(configured interface{}) *containerBeta.AddonsConf
...
+		ac.IstioConfig = &containerBeta.IstioConfig{
+			Disabled:        addon["disabled"].(bool),
+			Auth:            addon["auth"].(string),
+			ForceSendFields: []string{"Disabled", "Auth"},
+		}
+	}
+
+	if v, ok := config["cloudrun_config"]; ok && len(v.([]interface{})) > 0 {
+		addon := v.([]interface{})[0].(map[string]interface{})
+		ac.IstioConfig = &containerBeta.IstioConfig{
```
Comment (jcxplorer) [inline google/resource_container_cluster.go]: Not familiar with what cloudrun refers to, but is this supposed to be modifying the IstioConfig here?

Comment (pondohva) [reply google/resource_container_cluster.go]: you're right. CloudRun is a feature on top of Istio (CloudRun depends on Istio in container/v1beta1 API). Fixed, thank you

**Comment by pondohva** [comment]
@danawillow finally changed base, thanks.

**Comment by pondohva** [comment]
Also, I didn't run tests yet (go test without TF_ACC seems not so useful). Should I run integration tests with TF_ACC=1 on my own google cloud account?

**Comment by ndmckinley** [comment]
To answer your question, in general, yes, the way to make sure that you're submitting working code is to run the tests against your own account.  We'll run the tests too, but the debugging loop can be a bit frustrating if all you have is our test results (and can't look at logs or retry it).

**Comment by pondohva** [comment]
@ndmckinley thank you for answers and help with tests. I tested this on GKE (only TestAccContainerCluster_withIstioEnabled), abit longer than I expected but acceptance tests look really great. I need to move this PR to terraform-provider-google-beta according to https://www.terraform.io/docs/providers/google/provider_versions.html#google-beta. I can do it after you test everything.

**Comment by ndmckinley** [comment]
This PR needs to be moved to the beta provider; we have a robot that does that.  Consequently, closing this PR - you can follow along at the referenced https://github.com/GoogleCloudPlatform/magic-modules/pull/1101.

**Pull request closed without merging**